src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpContentReadStream.cs

```diff
@@ -18,7 +18,7 @@ public HttpContentReadStream(HttpConnection connection) : base(connection)
             {
             }
 
-            public sealed override bool CanRead => true;
+            public sealed override bool CanRead => _disposed == 0;
```

Comment:
If the goal here is increased consistency, then a similar change should be made in HttpContentWriteStream.CanWrite:
https://github.com/dotnet/runtime/blob/5996c03a44cb118e53bed8d542427d30f2094f40/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpContentWriteStream.cs#L20
(using `_connection != null` as an indication of not being disposed).

Reply:
And also in RawConnectionStream. And maybe some others.

As noted in the issue, I'm not convinced these changes are necessary: one shouldn't be using these objects after they're disposed. But I doubt the changes hurt much, so I'm ok with it, as long as the change is made comprehensively in this component, such that it improves rather that hurts consistency.

Reply:
Thanks. I added the missing fixes to the PR.